tests/std/tests/P0896R4_views_join/test.cpp

```diff
@@ -471,6 +471,41 @@ struct Immovable {
     Immovable& operator=(Immovable&&) = delete;
 };
 
+// Validate that the _Defaultabox primary template works when fed with a non-trivially-destructible type
+struct non_trivially_destructible_input_iterator {
+    constexpr ~non_trivially_destructible_input_iterator() {}
+
+    // When the type is default constructible, the partial specialization is used.
+    // Avoid that so we can test what we want to test.

```

Comment:
I took me a bit to realize that "the partial specialization" here is talking about `_Defaultabox`. Maybe "To test the correct specialization of _Defaultabox, this type must not be default constructible." would be more direct? (The comment in `P0896R4_views_single` could use a similar rephrasing.)